### ISSUE TITLE
The OneMoreAddIn installer fails halfway through the installation.

### ISSUE THREAD

**Issue by dj--jg** [stevencohn/OneMore#941]
# Problem to Solve

When running the OneMoreAddIn installer as an administrator, the installer fails halfway through the installation with the following message: 
'There is a problem with this Windows Installer package. A program run as part of the setup did not finish as expected. Contact your support personnel or package vendor.'

# To Reproduce

Probably not applicable because I assume this works for everyone else, but I just downloaded the installer (x64), ran the installer after the download finished, pressed next/okay/install, okayed the UAC prompt and halfway through the install it crashes each time.

# Expected Behavior

A succesfull installation

# Screenshots

https://i.imgur.com/4FJF1Wf.png
https://i.imgur.com/R0NKEsR.png

# Environment (if applicable)

 - OneNote Version: Onenote version is Version 2212 (Build 15928.20282), installed with an Office 365 subscription, afaik 64bit version.
 - OneMore Version 5.7.2 (or the installer for 5.7.2 anyway)

# Additional Context

I have enabled logging for the installer through regedit as mentioned in the wiki and uploaded the generated error log to this pastebin: https://privatebin.net/?e613cbef68c9fff0#GP3iV35op9dgqmE8Y2jLsCSr37T1UpzaHfQkVz5PkyWN

I can't make heads or tails from it, I hope someone else can.

**Comment by stevencohn** [comment]
Thank you for providing the msi logs. I think that really helps. Well, it helps me anyway. The suspicious entry I believe is this

`MSI (s) (20:84) [21:23:24:919]: Product: OneMoreAddIn -- Error 1722. There is a problem with this Windows Installer package. A program run as part of the setup did not finish as expected. Contact your support personnel or package vendor.  Action _67AC4AD8_6F00_4853_A52D_C3B81180AEFA, location: C:\Program Files\River\OneMoreAddIn\OneMoreSetupActions.exe, command: --install --x64`

Unfortunately, error 1722 is not at all specific. [This article might help](https://www.makeuseof.com/ways-fix-error-1722-windows-installer-package/). Please let me know if anything in that does help as I would like to either add it to the install guide, troubleshooting guide, or even try to overcome it in the installer itself if possible.

**Comment by dj--jg** [comment]
Okay, so I'll add some more mystery:

None of the steps in the article seemed to help, I tried #2 and #4, #3 didn't find any problems and didn't allow me to select the installer.

After doing all of the steps, I tried the 5.7.2 installer again and it failed. I then decided I would do one more hail mary and downloaded a couple of older releases, 5.7.0 and 4.6.

I tried 5.7.0 first and it installed just fine. Here is the log: https://privatebin.net/?d7fc28803637ef1b#7ciQ6d1JPwHhKk5CR9xFKYHe4nMHLRNvzv9y4m1ichWU

I then tried the 5.7.2 installer again and it still failing. I then tried 5.7.1 and it also failed.

I then got the bright idea to check if it is actually working, and I can't see any proof of the plugin working inside Onenote. Looking in the options menu, it is listed as an Inactive Application Add-In. Investigating further gives me this: 
![image](https://user-images.githubusercontent.com/5738606/222272328-15e139d9-9d62-4200-8785-d573436fd57d.png)

I assume the install probably didn't actually succeed.

I'll reboot once again now to see if that changes anything, if it doesn't I'm off to the pub and I'll look into doing a fresh Office install tomorrow.

**Comment by dj--jg** [comment]
Further update:

A bunch of files did end up in the right installation folder (OneMoreAddIn in Program Files), so I tried looking around there.

The calendar worked, but crashed when I tried to look at February. Then it suddenly worked fine the next time I looked at February. I manually ran the OneMoreSetupActions, but it didn't make the plugin suddenly work. It did however create a log file, and I also found a log file from SetupActions hidden inside SystemTemp, which is probably from the MSI installer running it during the installation?

Here is the log for OneMoreCalendar: https://privatebin.net/?2e05fe4dfa79cacb#EztWAVCKFXMctLYGXuRJvMf1XX85uaDzm7L92PhnEK2b

For OneMoreSetupActions inside SystemTemp: https://privatebin.net/?1f8facb725b059b2#AkcYu9npHCNAZxYQeK3yg6YaCTVM7GvEzmENkkqVo7Kx

And for OneMoreSetupActions inside Appdata/Local/Temp: https://privatebin.net/?e1f7710544a5eeee#7G8xtPn4MDNQuRrk3ErShUKkibFoGyWLVVoSR47HVLnf

**Comment by stevencohn** [comment]
So first I'm disappointed that the installer didn't rollback entirely and left at least some Registry entries and files hanging around after it failed. But that's another topic for another day.

The last log points to this [bit of code](https://github.com/stevencohn/OneMore/blob/main/OneMoreSetupActions/Actions/ShutdownOneNoteAction.cs#L70) which was introduced in 5.7.1 which would explain why 5.7.0 seemed to work for you.

So there's something about that WMI query that your laptop doesn't like. I'll need to do some digging myself but this is a good lead. Now go to the pub!

**Comment by dj--jg** [comment]
To be clear, the installer for 5.7.0 appears to finish cleanly, the plugin just doesn't work afterwards. It does however seem to suddenly finish while the loading bar is at the same point as where the 7.5.2 installer fails...

________________________________

**Comment by Jazzster56** [comment]
I'm having the same problem, but I don't get the message mentioned above when installing, but I do get it when I try to "repair" the installation. I installed this addon at my other location, but not sure which version it was. It was a couple of weeks ago. Both computers are Dell desktops running same version of Windows.

**Comment by franciscoyira** [comment]
> Are these all _udpate_ failures or are any of them first-time installations?

for me it is first-time installation

**Comment by stevencohn** [comment]
I feel awful that this is not working for some people. The latest versions 5.8.3 and 5.8.4 have been downloaded almost 2000 times in the last couple of days and so far, this installation failure seems to be affecting only a couple of people. But that's a couple too many.

I've tried a number of configurations over the past couple of days, both Win10 and Win11, Home and Pro, activated and non-activated, all with vanilla installations and zero customization. Install Windows. Install OneNote. Install **OneMore 5.8.4**. The bad news is that it worked successfully in all cases.

I do not know what the common denominator is in the failed case, if there is in fact a common denominator.

My only hope would be that I could find something of interest in the installation logs. So this is a plea to anyone unable to install OneMore to try the following experiment and capture some logs for me.

1. Open the Registry Editor as an administrator
    a. Create this key if not already there: `HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows\Installer`.
    b. Create a new String property named `Logging` with the value `voicewarmupx`
    c. That will enable verbose logging for MSI installers; you can remove the `Logging` property after this experiment
1. Install [OneMore 5.8.4](https://github.com/stevencohn/OneMore/releases/latest)
1. When the installation has failed and closed, gather these logs, or any of these that exist; tthese might be in %TEMP% or %TMP% depending on your system
    a. The MSI install log with a name similar to MSI1234.log. There may be multiple files like this from other installations; make sure you have the most recent one.
    b. %TEMP%\OneMoreSetup.log
    c. %TEMP%\OneMore.log
1.  Wrap them up in a Zip file or individually post them here or on a file share service and let me know

Thank you for your help, patience, and understanding. I'm trying to make OneMore work for everyone.

**Comment by dj--jg** [comment]
I just tried again with 5.8.4, it's saying that the installation succeeds but the plugin does not run in Onenote, still the same runtime error occurred message. I'll try your steps for extra logging this afternoon.

For now, here is the OneMoreSetup.log

<details>
======================================================================
starting action: --install .. 3/1/2023 10:50:34 PM
Installer architecture (x64), OS architecture (x64), requesting (x64)
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

Register... version 5.7.0

ShutdownOneNoteAction.Install ---
ONENOTE process not found

ProtocolHandlerAction.Install ---
step 0: opening HKLM:\Software\Classes\onemore
step 1: setting properties of HKLM:\Software\Classes\onemore
step 2: opening HKLM:\Software\Classes\onemore\shell\open\command
step 3: setting properties of HKLM:\Software\Classes\onemore\shell\open\command
verified: key created HKEY_LOCAL_MACHINE\Software\Classes\onemore\shell\open\command

TrustedProtocolAction.Install ---
translating user DESKTOP-86TH9JR\system to SID
DESKTOP-86TH9JR\system identity not mapped, falling back
fallback to search username in HKEY_USERS
defaulting Office version to 16
step 4: opening HKCU:\Software\Policies\Microsoft\Office\16.0\Common\Security\Trusted Protocols\All Applications\onemore:
install failed; error registering
System.ArgumentNullException
Message: Value cannot be null.
Parameter name: name
Source: mscorlib
StackTrace:    at System.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at Microsoft.Win32.RegistryKey.ValidateKeyName(String name)
   at Microsoft.Win32.RegistryKey.OpenSubKey(String name, Boolean writable)
   at OneMoreSetupActions.TrustedProtocolAction.Install()
   at OneMoreSetupActions.Program.Install()
TargetSite: [mscorlib] System.ThrowHelper::ThrowArgumentNullException()


======================================================================
starting action: --install .. 3/7/2023 11:00:09 PM
Installer architecture (x64), OS architecture (x64), requesting (x64)
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

Register... version 5.7.0

ShutdownOneNoteAction.Install ---
ONENOTE process not found

ProtocolHandlerAction.Install ---
step 0: opening HKLM:\Software\Classes\onemore
step 1: setting properties of HKLM:\Software\Classes\onemore
step 2: opening HKLM:\Software\Classes\onemore\shell\open\command
step 3: setting properties of HKLM:\Software\Classes\onemore\shell\open\command
verified: key created HKEY_LOCAL_MACHINE\Software\Classes\onemore\shell\open\command

TrustedProtocolAction.Install ---
translating user DESKTOP-86TH9JR\system to SID
DESKTOP-86TH9JR\system identity not mapped, falling back
fallback to search username in HKEY_USERS
defaulting Office version to 16
step 4: opening HKCU:\Software\Policies\Microsoft\Office\16.0\Common\Security\Trusted Protocols\All Applications\onemore:
install failed; error registering
System.ArgumentNullException
Message: Value cannot be null.
Parameter name: name
Source: mscorlib
StackTrace:    at System.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at Microsoft.Win32.RegistryKey.ValidateKeyName(String name)
   at Microsoft.Win32.RegistryKey.OpenSubKey(String name, Boolean writable)
   at OneMoreSetupActions.TrustedProtocolAction.Install()
   at OneMoreSetupActions.Program.Install()
TargetSite: [mscorlib] System.ThrowHelper::ThrowArgumentNullException()


======================================================================
starting action: --install .. 3/28/2023 10:13:27 AM
Installer architecture (x64), OS architecture (x64), requesting (x64)
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

Register... version 5.8.4

ShutdownOneNoteAction.Install ---
failed to run termination step of StopHost()
System.Management.ManagementException
Message: Invalid class 
Source: System.Management
StackTrace:    at System.Management.ManagementException.ThrowWithExtendedInfo(ManagementStatus errorCode)
   at System.Management.ManagementObjectCollection.ManagementObjectEnumerator.MoveNext()
   at System.Management.ManagementObjectCollection.get_Count()
   at OneMoreSetupActions.ShutdownOneNoteAction.StopHost()
TargetSite: [System.Management] System.Management.ManagementException::ThrowWithExtendedInfo()

install completed suspiciously

--------------------------------------------------
reentry action: --install-registrywow .. 3/28/2023 10:13:28 AM
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

RegistryWowAction.Install --- x64:True
1st chance, could not read CLSID or key missing
2nd chance, read CLSID ({DC67E480-C3CB-49F8-8232-60B0C2056C8E})
read path from LocalServer32 (C:\Program Files\Microsoft Office\Root\Office16\ONENOTE.EXE)
detected 64-bit install
WOW cloning not required

--------------------------------------------------
reentry action: --install-activesetup .. 3/28/2023 10:13:28 AM
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

ActiveSetupAction.Install ---
replacing '5,8,4' with '5,8,4'

======================================================================
starting action: --install .. 3/28/2023 10:14:32 AM
Installer architecture (x64), OS architecture (x64), requesting (x64)
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

Register... version 5.8.4

ShutdownOneNoteAction.Install ---
failed to run termination step of StopHost()
System.Management.ManagementException
Message: Invalid class 
Source: System.Management
StackTrace:    at System.Management.ManagementException.ThrowWithExtendedInfo(ManagementStatus errorCode)
   at System.Management.ManagementObjectCollection.ManagementObjectEnumerator.MoveNext()
   at System.Management.ManagementObjectCollection.get_Count()
   at OneMoreSetupActions.ShutdownOneNoteAction.StopHost()
TargetSite: [System.Management] System.Management.ManagementException::ThrowWithExtendedInfo()

install completed suspiciously

--------------------------------------------------
reentry action: --install-registrywow .. 3/28/2023 10:14:33 AM
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

RegistryWowAction.Install --- x64:True
1st chance, could not read CLSID or key missing
2nd chance, read CLSID ({DC67E480-C3CB-49F8-8232-60B0C2056C8E})
read path from LocalServer32 (C:\Program Files\Microsoft Office\Root\Office16\ONENOTE.EXE)
detected 64-bit install
WOW cloning not required

--------------------------------------------------
reentry action: --install-activesetup .. 3/28/2023 10:14:33 AM
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

ActiveSetupAction.Install ---
replacing '5.8.4' with '5,8,4'
</details>

It does say 'install completed suspiciously', so I guess it's not quite installing correctly.

For what it's worth, manually launching the calender application works fine, it also loads all the data from Onenote correctly.

Don't beat yourself up too much, it might be that there is nothing wrong with Onemore and just a very esoteric configuration issue. I'm leaning towards the latter because I do far too many weird configuration things with this PC, so I'll be trying a fresh install on an external SSD. If that helps, I might just do a fresh OS install on this PC.

I love Onemore on my other PC, thank you for all your development work :)

**Comment by stevencohn** [comment]
It looks to me like you're logged in as the SYSTEM account. Can that be true? I thought that was a non-interactive account but you did say that you have "many weird configuration things" so I have to ask. The SYSTEM account is basically a super user but is also weird itself in some ways.

If so, create another local account with admin privileges and try running it from that account.

This is what a normal setup log looks like. Compare the TrustedProtocolAction.Install section below with your above.

<details>
======================================================================
starting action: --install .. 3/28/2023 6:43:55 AM
Installer architecture (x64), OS architecture (x64), requesting (x64)
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

Register... version 5.8.3

ShutdownOneNoteAction.Install ---
dllhost process not found
ONENOTE process not found

ProtocolHandlerAction.Install ---
step 0: opening HKLM:\Software\Classes\onemore
step 1: creating HKLM:\Software\Classes\onemore
step 2: setting properties of HKLM:\Software\Classes\onemore
step 3: opening HKLM:\Software\Classes\onemore\shell\open\command
step 4: creating HKLM:\Software\Classes\onemore\shell\open\command
step 5: setting properties of HKLM:\Software\Classes\onemore\shell\open\command
verified: key created HKEY_LOCAL_MACHINE\Software\Classes\onemore\shell\open\command

TrustedProtocolAction.Install ---
translating user CORP\usstckm to SID
registering trusted protocol for user CORP\usstckm (S-1-5-21-2724133749-1153703018-3936594150-7829)
defaulting Office version to 16
step 6: opening HKCU:\Software\Policies\Microsoft\Office\16.0\Common\Security\Trusted Protocols\All Applications\onemore:
step 7: creating HKCU:\Software\Policies\Microsoft\Office\16.0\Common\Security\Trusted Protocols\All Applications\onemore:
key created HKEY_USERS\S-1-5-21-2724133749-1153703018-3936594150-7829\Software\Policies\Microsoft\Office\16.0\Common\Security\Trusted Protocols\All Applications\onemore:

EdgeWebViewAction.Install ---
WebView2 Runtime is already installed
install completed successfully

--------------------------------------------------
reentry action: --install-registrywow .. 3/28/2023 6:43:56 AM
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

RegistryWowAction.Install --- x64:True
1st chance, could not read CLSID or key missing
2nd chance, read CLSID ({DC67E480-C3CB-49F8-8232-60B0C2056C8E})
could not read LocalServer32 or key missing
read path from WOW6432Node\..\LocalServer32 (C:\Program Files (x86)\Microsoft Office\Root\Office16\ONENOTE.EXE)
detected 32-bit install
step 0: deleting CLSID clone
step 1: cloning CLSID
step 2: copying from HKEY_CLASSES_ROOT\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61} to HKEY_CLASSES_ROOT\WOW6432Node\CLSID
created clone HKEY_CLASSES_ROOT\WOW6432Node\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}
clone property  = River.OneMoreAddIn.AddIn (String)
clone property AppID = {88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61} (String)
created subkey HKEY_CLASSES_ROOT\WOW6432Node\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\Implemented Categories
created subkey HKEY_CLASSES_ROOT\WOW6432Node\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\Implemented Categories\{62C8FE65-4EBB-45E7-B440-6E39B2CDBF29}
created subkey HKEY_CLASSES_ROOT\WOW6432Node\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\InprocServer32
clone property  = mscoree.dll (String)
clone property ThreadingModel = Both (String)
clone property Class = River.OneMoreAddIn.AddIn (String)
clone property CodeBase = C:\Program Files\River\OneMoreAddIn\River.OneMoreAddIn.dll (String)
clone property RuntimeVersion = v4.0.30319 (String)
clone property Assembly = River.OneMoreAddIn, Version=5.8.3.0, Culture=neutral, PublicKeyToken=null (String)
created subkey HKEY_CLASSES_ROOT\WOW6432Node\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\InprocServer32\5.8.3.0
clone property CodeBase = C:\Program Files\River\OneMoreAddIn\River.OneMoreAddIn.dll (String)
clone property RuntimeVersion = v4.0.30319 (String)
clone property Class = River.OneMoreAddIn.AddIn (String)
clone property Assembly = River.OneMoreAddIn, Version=5.8.3.0, Culture=neutral, PublicKeyToken=null (String)
created subkey HKEY_CLASSES_ROOT\WOW6432Node\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\ProgID
clone property  = River.OneMoreAddin (String)
created subkey HKEY_CLASSES_ROOT\WOW6432Node\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\Programmable
clone property  =  (String)
created subkey HKEY_CLASSES_ROOT\WOW6432Node\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\TypeLib
clone property  = {88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61} (String)
created subkey HKEY_CLASSES_ROOT\WOW6432Node\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\VersionIndependentProgID
clone property  = River.OneMoreAddIn (String)

--------------------------------------------------
reentry action: --install-activesetup .. 3/28/2023 6:43:56 AM
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

ActiveSetupAction.Install ---
replacing '5.8.3' with '5,8,3'
</details>

**Comment by kestutiz** [comment]
got same error, let's look:
<details>
======================================================================
starting action: --install .. 2023-05-18 23:39:56
Installer architecture (x64), OS architecture (x64), requesting (x86)
Installer architecture (x64) does not match OS (x64) or request (x86)
======================================================================
</details>
Bit confused about instructions and file naming. I'm with win10 64bit and office 32 bit. Successfully installed x64 version. extract:
<details>
starting action: --install .. 2023-05-18 23:41:10
Installer architecture (x64), OS architecture (x64), requesting (x64)
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

Register... version 5.8.4

ShutdownOneNoteAction.Install ---
dllhost process not found
ONENOTE process not found

ProtocolHandlerAction.Install ---
step 0: opening HKLM:\Software\Classes\onemore
step 1: creating HKLM:\Software\Classes\onemore
step 2: setting properties of HKLM:\Software\Classes\onemore
step 3: opening HKLM:\Software\Classes\onemore\shell\open\command
step 4: creating HKLM:\Software\Classes\onemore\shell\open\command
step 5: setting properties of HKLM:\Software\Classes\onemore\shell\open\command
verified: key created HKEY_LOCAL_MACHINE\Software\Classes\onemore\shell\open\command

TrustedProtocolAction.Install ---
translating user xxxxx to SID
registering trusted protocol for user xxxxx (S-1-5-21-1081476967-3755237250-1015573492-1001)
defaulting Office version to 16
step 6: opening HKCU:\Software\Policies\Microsoft\Office\16.0\Common\Security\Trusted Protocols\All Applications\onemore:
step 7: creating HKCU:\Software\Policies\Microsoft\Office\16.0\Common\Security\Trusted Protocols\All Applications\onemore:
key created HKEY_USERS\S-1-5-21-1081476967-3755237250-1015573492-1001\Software\Policies\Microsoft\Office\16.0\Common\Security\Trusted Protocols\All Applications\onemore:

EdgeWebViewAction.Install ---
WebView2 Runtime is already installed
install completed successfully

--------------------------------------------------
reentry action: --install-registrywow .. 2023-05-18 23:41:11
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

RegistryWowAction.Install --- x64:True
1st chance, could not read CLSID or key missing
2nd chance, read CLSID ({D7FAC39E-7FF1-49AA-98CF-A1DDD316337E})
read path from LocalServer32 (C:\PROGRA~2\MICROS~1\Office14\ONENOTE.EXE)
detected 64-bit install
WOW cloning not required

--------------------------------------------------
reentry action: --install-activesetup .. 2023-05-18 23:41:11
OneMore installer running as user NT AUTHORITY\SYSTEM (S-1-5-18) elevated

ActiveSetupAction.Install ---
replacing '5.8.4' with '5,8,4'
</details>

**Comment by MrNobdyyy** [comment]
I got the same error, but there is a way to solve the problem:
right click the .msi file, and click "properties", and tick the "Run this program in compatibility mode for", here is the picture.
(my English is bad...)
![uTools_1690520507459](https://github.com/stevencohn/OneMore/assets/35237425/117a9212-1583-47a1-8842-c715185ea383)

**Comment by stevencohn** [comment]
Since I've been unable to reproduce the issue myself, it would be great if someone who's observed the failure could try @MrNobdyyy 's suggestion. 

1. Right-click the .msi
2. Choose Properties
3. Click the Compatibility tab
4. Tick the box _Run this program in compatibility mode for_
5. Choose _Previous version of Windows_
6. Click OK
7. Double-clcik the .msi

Thank you @MrNobdyyy

**Comment by Maddyson1** [comment]
Good morning,
I am also a victim of this error message. Sadness.
It is still present despite the compatibility change.
New solutions would be welcome ^^

**Comment by badrelmers** [comment]
I think I found the problem, this can happen for 2 things in my case with win7:

1.  Use the x64 installer when running on a 64-bit Windows machine. I have onenote x32 on win7 x64 and the new Onemore x32 failed to install, you have to install Onemore x64 even if you use onenote x32 because the OS is x64.

2. open your firewall and make sure you have internet because Onemore need to download something called "Microsoft Edge WebView2 Evergreen Bootstrap". if you do not have internet or you do not want to install that thing then follow this steps:

OneMore_3.27 worked always fine for me but OneMore_5.8.5 want to download and install  from internet "Microsoft Edge WebView2": WebView2 Evergreen Bootstrap

to bypass this installation i run this to make Onemore think that Edge WebView2 is already installed, run this as admin:

```batch
reg add "HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\EdgeUpdate\Clients" /v "RuntimeId" /t REG_SZ /d "{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}" /f
reg add "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\EdgeUpdate\Clients" /v "RuntimeId" /t REG_SZ /d "{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}" /f

reg add "HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\EdgeUpdate\Clients\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}" /f
reg add "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\EdgeUpdate\Clients\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}" /f
reg add "HKEY_CURRENT_USER\Software\Microsoft\EdgeUpdate\Clients\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}" /f
 ```

i got those entries from this files
```
https://github.com/stevencohn/OneMore/blob/77a20c2c8c91159a4dc75eb58612ecc00b35e53f/iq.ps1#L307
https://github.com/stevencohn/OneMore/blob/77a20c2c8c91159a4dc75eb58612ecc00b35e53f/OneMoreSetupActions/Actions/EdgeWebViewAction.cs#L23
```

with that, Onemore installs successfully without installing Edge WebView2, and everything works fine in Onemore!

**Comment by stevencohn** [comment]
Yes, not installing WebView2 is safe; it will break the Import from Web command but that's about it. This is likely one cause but may not be all of them because WebView2 comes pre-installed on Win11 and Win11 users have reported similar installer failures.

But I can at least check whether there is a viable internet connection and, if not, skip the WebView2 action to allow the installer to continue successfully. I'll investigate...

**Comment by dj--jg** [comment]
I just installed 5.9.0 x64 and the install completed flawlessly, I'm very happy I can now also use Onemore on my second laptop, especially since this is the laptop I use most with Onenote. Thank you very much :)

**Comment by RandyR34** [comment]
Here is a guide to help resolve the issue.  I have ran into this bug many times.  Short answer is, need to run the installer "as administrator".  However, one cannot right click on an MSI file to "run as administrator", so the work-a-round it to run it from the command line instead.    

### First, example of the installer failing:   
![image](https://github.com/stevencohn/OneMore/assets/15321147/fa31670f-5f3f-41a1-9c43-2c6cd42d9421)

![image](https://github.com/stevencohn/OneMore/assets/15321147/bed06aa8-95bd-4149-8f1b-4d8f535e6e4c)

![image](https://github.com/stevencohn/OneMore/assets/15321147/cf0448ed-be52-4d05-b898-563a02e2ebae)

### To resolve: 
1. Save the MSI installer file inside a dummy directory C:\1_DeleteMe
2. Click the Windows start button in the lower left-hand corner of the screen.  Then type "cmd" to find this "command prompt" app.  **Important...open the command prompt as administrator!**  
![image](https://github.com/stevencohn/OneMore/assets/15321147/95c1b416-e551-47de-a32b-49693c4a7c2a)
3. Run the command prompt as shown below.  
![image](https://github.com/stevencohn/OneMore/assets/15321147/69e1ea34-a3e7-4927-a37e-6a483c017af7)
4. Wait for installer to finish.  The installer should complete successfully.

**Comment by stevencohn** [comment]
Hi. Thanks for this.

The installation log file %temp%\onemoresetup.log should report its elevation level right at the beginning, something like "OneMore installer running as user {username} ({sid}) *elevated*"

On your system, does it not say "elevated" at the end of that line when you double-click the installer? Are you logged in as a normal user, non-admistrator?

**Comment by stevencohn** [comment]
I'll look into aborting the installation if not an admin and display a message to the user.

**Comment by RandyR34** [comment]
Hey Steve,
Just mentioning this for breadcrumb sake.  No need to reopen the issue.  

1. A co-worker, with local admin rights, performed the initial installation by double-clicking the MSI installer.  However, as shown below, the menu items were incomplete.  We also tried removing/adding back the addin via OneNote->File->Options->Addins but it had no effect.   
![Image](https://github.com/user-attachments/assets/744383ba-8229-4daf-9e0a-909318c25e15)

2. Then we performed a repair of the OneMore installation.  The repair action helped the menu items display correctly.  However, clicking on the menu items (to insert a code snippet etc) did not perform any action in OneNote.  Aka the menu items displayed correctly, but clicking on them did nothing.  

![Image](https://github.com/user-attachments/assets/45ce0c99-8bf0-449e-9cc7-541bcfdeaa70)

3. Ultimately, we used the "Install as administrator using the command line trick" mentioned on this same thread [here](https://github.com/stevencohn/OneMore/issues/941#issuecomment-2083182784).  This resolved the issue.
